### PR TITLE
Fix performance issues from 1.x

### PR DESCRIPTION
**Pull request by lbeschastny**
This PR fixes some issues introduced by the resent major release due to incorrectly filtered traces:

 * Huge dependency deterioration
 * Non-existing trace ids being propagated to downstream services in some edge cases

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -44,12 +44,10 @@ rstest = "0.18"
 
 [features]
 full = ["test"]
-default = ["otlp", "zipkin"]
+default = ["zipkin"]
 zipkin = ["dep:opentelemetry-zipkin"]
-otlp = ["opentelemetry-otlp/http-proto", "tracer", "dep:tracing-opentelemetry-instrumentation-sdk", "dep:opentelemetry-http", "dep:hyper"]
```
Comment (lbeschastny) [inline Cargo.toml:49]: I decided to remove this feature since the crate is not usable without it

File: src/lib.rs [hunk 2]
```diff
@@ -151,11 +140,10 @@ pub fn init_tracing_with_fallbacks(
         propagation::TextMapSplitPropagator::from_env().expect("setup of Propagation"),
     );
 
-    let otel_layer = tracing_opentelemetry::layer()
-        .with_tracer(otel_tracer)
-        .with_filter(filter::OtelFilter::default());
+    let otel_layer = tracing_opentelemetry::layer().with_tracer(otel_tracer);
     let subscriber = tracing_subscriber::registry()
-        .with(build_logger_text(log_level))
+        .with(Into::<filter::TracingFilter>::into(log_level))
```
Comment (lbeschastny) [inline src/lib.rs:145]: This line is the actual fix.

But I also had to rewrite our custom `TracingFilter` struct to be able to use it here like that.

Comment (lbeschastny) [reply src/lib.rs:145]: Previously the filtering was done in the `otel_layer` layer, but it's not correct

**Review by Travispersson — APPROVED** [review]
🚀

**Review by jaypadma — APPROVED** [review]
